### ISSUE TITLE
Rename "soft deleted" objects when users 'claim' a net/org/fac/car/ix name of a soft-deleted object

### ISSUE THREAD

**Issue by netravnen** [peeringdb/peeringdb#1559]
**Is your feature request related to a problem? Please describe.**
Rename soft-deleted objects when a user tries to 'claim' an object name of a soft-deleted object.

**Who is affected by the problem?**
Users hitting the error 'name already exists/name already exists'

**What is the impact?**
More accurate object names for objects with status 'ok'

**Are there security concerns?**
None

**Are there privacy concerns?**
None

**Describe the solution you'd like**
Renamed the soft-deleted object, append the soft-deleted objects id to its name. And allow the object with the status of 'ok' to take its name.

```
E.g. name: A, status: deleted,
name: B, status: ok,
name_change: A->A #1, B->A

```
**Do you think this feature will require a formal design?**
no

**Describe alternatives you've considered**
None

**Could this feature request need support from the Admin Committee?**
None

**What is the proposed priority?**
Simply urgent

**Provide a rationale for any/all of the above**
Allow users to "reclaim" object names of deleted objects to avoid ("lessen") name-space collisions

**Additional context**
None

**Comment by jackcarrozzo** [comment]
PC call: undeleting does happen sometimes, and a collision could occur there. More discussion required.

**Comment by jackcarrozzo** [comment]
PC call: admins can rename the obj - Arnold uses three stars.

**Comment by netravnen** [comment]
> PC call: admins can rename the obj

@jackcarrozzo True. AC can rename obj. ⟶ Can you explain why I should then receive a user-generated ticket for a (IMO) low-hanging fruit that can lessen the amount of tickets AC receive?

**Comment by martinhannigan** [comment]
AC can use all the automation we can send to it. I’m +1 (missed call or
would’ve added color).

Thanks

**Comment by arnoldnipper** [comment]
> What would be your argument _against_ automating the above suggestion. That has potential to lessen the amount of tickets AC receive? (Albeit, I admit, the number is low, but there are still some)

We do merger and unmerger. If you rename automatically that wouldn't work anymore. And name collisions don't happen that often. Hence, I would still leave it to AC to decide case by case together with the customer(s)

**Comment by netravnen** [comment]
We already append the OBJ-id to the name field when a user deletes e.g. a FAC. I.e. status changes from PENDING to DELETED (object was not approved before the user deleted it).

Which leads me we already have a logic that partially does what was initially described in this issue.

**Comment by netravnen** [comment]
https://www.peeringdb.com/cp/peeringdb_server/facility/15345/history/

https://www.peeringdb.com/cp/peeringdb_server/facility/15247/history/

**Comment by leovegoda** [comment]
Closing as there's been no comment in over a year
